app/src/main/java/com/epfl/beatlink/ui/search/components/SearchBars.kt

```diff
@@ -0,0 +1,180 @@
+                            ),
+                        modifier =
+                            Modifier.testTag("nonWritableSearchBarText")
+                                .padding(start = 8.dp, top = 4.dp, bottom = 4.dp))
+                  }
+            }
+      }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ShortSearchBar() {
+  var searchQuery = remember { mutableStateOf(TextFieldValue("")) }
+  val focusRequester = remember { FocusRequester() }
+  val focusManager = LocalFocusManager.current
+
+  // Automatically request focus when the composable is first loaded
+  LaunchedEffect(Unit) { focusRequester.requestFocus() }
+
+  OutlinedTextField(
+      value = searchQuery.value,
+      onValueChange = { searchQuery.value = it },
+      placeholder = {
+        Text(
+            text = "",
+            style =
+                TextStyle(
+                    fontSize = 16.sp,
+                    lineHeight = 20.sp,
+                    fontWeight = FontWeight(400),
+                    color = Color(0xFF6F6F6F),
+                    letterSpacing = 0.16.sp,
+                ),
+            modifier = Modifier.testTag("writableSearchBarText").padding(top = 4.dp, bottom = 4.dp))
+      },
+      singleLine = true,
+      leadingIcon = {
+        Icon(
+            imageVector = Icons.Default.Search,
+            contentDescription = "Search Icon",
+            tint = Color(0xFF6F6F6F),
+            modifier =
+                Modifier.testTag("writableSearchBarIcon")
+                    .size(28.dp)
+                    .padding(start = 5.dp, top = 4.dp, bottom = 4.dp))
+      },
+      colors =
+          TextFieldDefaults.outlinedTextFieldColors(
+              focusedBorderColor = Color.Transparent, // Color when selected (focused)
+              unfocusedBorderColor = Color.Transparent, // Color when not selected
+              cursorColor = Color.Black),
+      shape = RoundedCornerShape(5.dp),
+      modifier =
+          Modifier.testTag("writableSearchBar")
+              .focusRequester(focusRequester)
+              .shadow(
+                  elevation = 4.dp, spotColor = Color(0x26000000), ambientColor = Color(0x26000000))
+              .border(width = 1.dp, color = Color.White)
+              .width(334.dp)
+              .height(48.dp)
+              .border(
+                  width = 1.dp, color = Color.Transparent, shape = RoundedCornerShape(size = 5.dp))
+              .background(color = Color(0xFFF2F2F2), shape = RoundedCornerShape(size = 5.dp)))
+}
+
+@Composable
+fun ShortSearchBarLayout(navigationActions: NavigationActions) {
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier =
+          Modifier.testTag("shortSearchBarRow")
+              .fillMaxWidth()
+              .padding(start = 21.dp, end = 8.dp, top = 6.dp, bottom = 6.dp)) {
+        // Back Icon
+        Icon(
+            painter = painterResource(id = R.drawable.back_arrow),
```

Comment:
you should have a look on how to use the already existing icons in material3 and painting them afterward instead of downloading a separate images each time